tests/foreman/api/test_host.py

```diff
@@ -1573,145 +1353,67 @@ def test_positive_read_puppet_ca_proxy_name(self):
 
         :CaseImportance: Critical
         """
-        interface = entities.Interface(host=self.host).create()
-        for new_name in valid_interfaces_list():
-            with self.subTest(new_name):
-                interface.name = new_name
-                interface = interface.update(['name'])
-                self.assertEqual(interface.name, new_name)
+        name = gen_choice(invalid_interfaces_list())
+        with pytest.raises(HTTPError) as error:
+            entities.Interface(host=module_host, name=name).create()
+            assert error.exception.response.status_code == 422
+        interface = entities.Interface(host=module_host).create()
+        interface.name = name
+        with pytest.raises(HTTPError) as error:
+            interface.update(['name'])
+            assert interface.read().name != name
+            assert error.exception.response.status_code == 422
 
-    @tier1
-    def test_negative_update_name(self):
-        """Attempt to update interface name with different invalid entries
-        (>255 chars, unsupported string types)
-
-        :id: 6a1fb718-adfb-47cb-b28c-fb3cd01f99b0
-
-        :expectedresults: An interface is not updated
-
-        :CaseImportance: Critical
-        """
-        interface = entities.Interface(host=self.host).create()
-        for new_name in invalid_interfaces_list():
-            with self.subTest(new_name):
-                interface.name = new_name
-                with self.assertRaises(HTTPError) as error:
-                    interface.update(['name'])
-                self.assertNotEqual(interface.read().name, new_name)
-                self.assertEqual(error.exception.response.status_code, 422)
-
-    @tier1
-    def test_positive_delete(self):
-        """Delete host's interface (not primary)
-
-        :id: 9bf83c3a-a4dc-420e-8d47-8572e5ae1dd6
-
-        :expectedresults: An interface is successfully deleted
-
-        :CaseImportance: Critical
-        """
-        host = entities.Host().create()
-        interface = entities.Interface(host=host).create()
-        interface.delete()
-        with self.assertRaises(HTTPError):
-            interface.read()
-
-    @tier1
-    def test_negative_delete(self):
-        """Attempt to delete host's primary interface
-
-        :id: 716a9dfd-0f31-45aa-a6d1-42add032a15c
-
-        :expectedresults: An interface is not deleted
-
-        :CaseImportance: Critical
-        """
-        host = entities.Host().create()
         primary_interface = next(
-            interface for interface in host.interface if interface.read().primary
+            interface for interface in module_host.interface if interface.read().primary
         )
-        with self.assertRaises(HTTPError):
+        with pytest.raises(HTTPError):
             primary_interface.delete()
-        with self.assertNotRaises(HTTPError, expected_value=404):
+        try:
+            HTTPError
```

Comment:
Is this intentionally trying to check HTTPError ? Whats the reason behind testing it ?

Reply:
now the reason is in the description 
You are unable to remove primary interface

Reply:
I don't understand this part of code, maybe that's just my lack of knowledge. But:

You have a try block. In that try block, you *instantiate* a new object of type HTTPError. That should always succeed unless you haven't imported it. That alone seems fishy.
Then, you try to catch an exception with a type of integer 404. Which is never thrown anywhere.

I think I know what you wanted to do but I doubt (without any real testing) that this code achieves it.

Reply:
thanks @lhellebr, this change was done by automatic tool unittest2pytest so I was not aware of the problem, 
No I am and sorry that's my mistake. I agree with you it does not make a sense. I removed it form test.

Reply:
DONE.